framework/src/main/java/org/checkerframework/framework/ajava/JointJavacJavaParserVisitor.java

```diff
@@ -0,0 +1,2382 @@
+    public Void visitDoWhileLoop(DoWhileLoopTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof DoStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, DoStmt.class);
+        }
+
+        DoStmt node = (DoStmt) javaParserNode;
+        processDoWhileLoop(javacTree, node);
+        // In javac the condition is parenthesized but not in JavaParser.
+        ParenthesizedTree condition = (ParenthesizedTree) javacTree.getCondition();
+        condition.getExpression().accept(this, node.getCondition());
+        javacTree.getStatement().accept(this, node.getBody());
+        return null;
+    }
+
+    @Override
+    public Void visitEmptyStatement(EmptyStatementTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof EmptyStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, EmptyStmt.class);
+        }
+
+        processEmptyStatement(javacTree, (EmptyStmt) javaParserNode);
+        return null;
+    }
+
+    @Override
+    public Void visitEnhancedForLoop(EnhancedForLoopTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof ForEachStmt)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, ForEachStmt.class);
+        }
+
+        ForEachStmt node = (ForEachStmt) javaParserNode;
+        processEnhancedForLoop(javacTree, node);
+        javacTree.getVariable().accept(this, node.getVariableDeclarator());
+        javacTree.getExpression().accept(this, node.getIterable());
+        javacTree.getStatement().accept(this, node.getBody());
+        return null;
+    }
+
+    @Override
+    public Void visitErroneous(ErroneousTree javacTree, Node javaParserNode) {
+        // An erroneous tree is a malformed expression, so skip.
+        return null;
+    }
+
+    @Override
+    public Void visitExports(ExportsTree javacTree, Node javaParserNode) {
+        if (!(javaParserNode instanceof ModuleExportsDirective)) {
+            throwUnexpectedNodeType(javacTree, javaParserNode, ModuleExportsDirective.class);
+        }
+
+        ModuleExportsDirective node = (ModuleExportsDirective) javaParserNode;
+        processExports(javacTree, node);
+        visitLists(javacTree.getModuleNames(), node.getModuleNames());
+        javacTree.getPackageName().accept(this, node.getName());
+        return null;
+    }
+
+    @Override
+    public Void visitExpressionStatement(ExpressionStatementTree javacTree, Node javaParserNode) {
+        if (javaParserNode instanceof ExpressionStmt) {
+            ExpressionStmt node = (ExpressionStmt) javaParserNode;
+            processExpressionStatemen(javacTree, node);
+            javacTree.getExpression().accept(this, node.getExpression());
+        } else if (javaParserNode instanceof ExplicitConstructorInvocationStmt) {
+            // In this case the expression will be a MethodTree, which would be better to match with
+            // the statement than the expression statement itself.
```

Comment:
I'm confused by this comment.  Should it be "the expression than the expression statement itself."?  In any event, please clarify.